nx.json

```diff
@@ -18,7 +18,9 @@
       }
     },
     "docs": {
```

Comment:
adding doc building to the pipeline.

This also runs tsc it seems. Basically if we "successfully build" docs in the CI we can choose to upload them somewhere in the future. 

Also just gives us a check against all the types of tools we are useing, make sure it works in a fresh env.